autoload/ale/lsp/window.vim

```diff
@@ -0,0 +1,68 @@
+" Author: suoto <andre820@gmail.com>
+" Description: Handling of window/* LSP methods, although right now only
+" handles window/showMessage
+
+" Constants for message type codes
+let s:LSP_MESSAGE_TYPE_ERROR = 1
+let s:LSP_MESSAGE_TYPE_WARNING = 2
+let s:LSP_MESSAGE_TYPE_INFORMATION = 3
+let s:LSP_MESSAGE_TYPE_LOG = 4
+
+" Translate strings from the user config to a number so we can check
+" severities
+let s:CFG_TO_LSP_SEVERITY = {
+\   'error': s:LSP_MESSAGE_TYPE_ERROR,
+\   'warning': s:LSP_MESSAGE_TYPE_WARNING,
+\   'information': s:LSP_MESSAGE_TYPE_INFORMATION,
+\   'log': s:LSP_MESSAGE_TYPE_LOG
+\}
+
+" This formats string 'a:format' by replacing a:args keys by their respective
+" values
+" - format: base format, where keys are surrounded by '%' (e.g, %linter%)
+" - args: dict defining values for each key inside format. Don't add extra '%'
+"   to the keys, they'll be added automatically when replacing
+function! ale#lsp#window#formatString(format, args) abort
+    let l:string = a:format
+
+    for [l:key, l:value] in items(a:args)
+        let l:string = substitute(l:string, '\V' . l:key, '\=l:value', 'g')
+    endfor
+
+    return l:string
+endfunction
+
+" Handle window/showMessage response.
+" - details: dict containing linter name and format (g:ale_lsp_show_message_format)
+" - params: dict with the params for the call in the form of {type: number, message: string}
+function! ale#lsp#window#HandleShowMessage(linter_name, format, params) abort
+    let l:message = a:params.message
+    let l:type = a:params.type
+
+    " Discard log severity for now
+    if l:type is# s:LSP_MESSAGE_TYPE_LOG
+        return
+    endif
```

Comment:
Because you have the setting for controlling the log level, you can show log level messages if the level is set that high, now that the default is to only show error messages.

Reply:
Ok, will also add a new config, `g:ale_echo_msg_log_str`, to configure the string